File: solutions/r/grains/3/grains.R
square <- function(n) {
	if (n == 0) stop ("square 0 raises an exception")
	if (n < 0) stop ("negative square raises an exception")
	if (n > 64) stop ("square greater than 64 raises an exception")

	g <- 2^(n - 1)

	return (g)
}

total <- function() {
	tg <- sum(square(1:64))
	return (tg)
}

File: solutions/c/grains/3/src/grains.c
#include "grains.h"
#include "ctype.h"
#include "stdio.h"

unsigned long long square(unsigned int n)
{
	if (n==0 || n > 64) return (0);
	unsigned long long g = 1;
	for (unsigned int i=2; i<=n; i++){
 		g *= 2;         
	}
	return g;
}

unsigned long long total()
{
	unsigned long long t = 0;
	for (unsigned int i=1; i<=64; i++){
 		t += square(i);         
	}
	return t;
}
